src/Network/HTTP/StackClient.hs

```diff
@@ -83,3 +96,52 @@ withResponse
   :: (MonadUnliftIO m, MonadIO n)
   => Request -> (Response (ConduitM i Strict.ByteString n ()) -> m a) -> m a
 withResponse = Network.HTTP.Simple.withResponse . setUserAgent
+
+-- | Set the user-agent request header
+setGithubHeaders :: Request -> Request
+setGithubHeaders = setRequestHeader "Accept" ["application/vnd.github.v3+json"]
+
+-- | Download the given URL to the given location. If the file already exists,
+-- no download is performed. Otherwise, creates the parent directory, downloads
+-- to a temporary file, and on file download completion moves to the
+-- appropriate destination.
+--
+-- Throws an exception if things go wrong
+download :: HasTerm env
+         => Request
+         -> Path Abs File -- ^ destination
+         -> RIO env Bool -- ^ Was a downloaded performed (True) or did the file already exist (False)?
+download req dest = Download.download (setUserAgent req) dest
+
+-- | Same as 'download', but will download a file a second time if it is already present.
+--
+-- Returns 'True' if the file was downloaded, 'False' otherwise
+redownload :: HasTerm env
+           => Request
+           -> Path Abs File -- ^ destination
+           -> RIO env Bool
+redownload req dest = Download.redownload (setUserAgent req) dest
+
+-- | Copied and extended version of Network.HTTP.Download.download.
+--
+-- Has the following additional features:
+-- * Verifies that response content-length header (if present)
+--     matches expected length
+-- * Limits the download to (close to) the expected # of bytes
+-- * Verifies that the expected # bytes were downloaded (not too few)
+-- * Verifies md5 if response includes content-md5 header
+-- * Verifies the expected hashes
+--
+-- Throws VerifiedDownloadException.
+-- Throws IOExceptions related to file system operations.
+-- Throws HttpException.
+verifiedDownload
```

Comment:
I realize this isn't something you changed, but already existed. But question: is there any reason we wouldn't _always_ want to perform a verified download instead of an unverified download? In other words, can we merge these two functions together?

Reply:
It looks like `download` is only used `loadGlobalHints`, in `Stack.New` we use `redownload` but that one looks to be not doing checks done in `verifiedDownload`. Maybe it's better to deal with this in a separate issue?

Reply:
Opened #4524 for that